crates/nargo/src/cli/prove_cmd.rs

```diff
@@ -43,6 +43,16 @@ fn process_abi_with_input(
     abi: Abi,
     witness_map: &BTreeMap<String, InputValue>,
 ) -> Result<(BTreeMap<Witness, FieldElement>, Option<Witness>), CliError> {
+    // Check that enough witness values were supplied
+    let num_params = abi.num_parameters();
+    if num_params != witness_map.len() {
+        return Err(CliError::Generic(format!(
+            "Expected {} number of values, but got {} number of values",
```

Comment:
Same comment as in the verifier_cmd
Also could you move it below the check for the parameter names, which generate a more informative error.
Then, since we would know the names are 'correct', you could display the parameter that is not in the ABI.